rocketpy/motors/motor.py

```diff
@@ -1224,7 +1224,9 @@ def exhaust_velocity(self):
         self.exhaust_velocity : Function
             Gas exhaust velocity of the motor.
         """
-        return self.total_impulse / self.propellant_initial_mass
+        return Function(
+            self.total_impulse / self.propellant_initial_mass
+        ).set_discrete_based_on_model(self.thrust)
```

Comment:
Do we need to return a Function here if this is already a `@funcify_method` ?

Reply:
The Function is called so we can discretize it. If we applied @funcify directly, the Function source would be a numerical value.

The main issue this PR tackles are hidden lambda based Functions that end up cascading the others in arithmetic, making code slow.